packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,308 @@
+}
+
+const selectBoxStyle = style({
+  ...focusRing(),
+  aspectRatio: {
+    orientation: {
+      vertical: 'square'
+    }
+  },
+  backgroundColor: {
+    default: 'white',
+    isDisabled: 'disabled'
+  },
+  borderWidth: 2,
+  borderStyle: {
+    default: 'solid',
+    isDisabled: 'none'
+  },
+  borderColor: {
+    default: 'gray-25',
+    isSelected: 'black'
+  },
+  borderRadius: 'lg',
+  boxShadow: 'elevated',
+  boxSizing: 'border-box',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  },
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  maxHeight: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(72),
+          M: size(106)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(168),
+          S: size(184),
+          M: size(200),
+          L: size(216),
+          XL: size(232)
+        }
+      }
+    }
+  },
+  minHeight: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(72),
+          M: size(106)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(100),
+          S: size(128),
+          M: size(144),
+          L: size(160),
+          XL: size(192)
+        }
+      }
+    }
+  },
+  maxWidth: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(168),
+          S: size(184),
+          M: size(200),
+          L: size(216),
+          XL: size(232)
+        }
+      }
+    }
+  },
+  minWidth: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(100),
+          S: size(128),
+          M: size(144),
+          L: size(160),
+          XL: size(192)
+        }
+      }
+    }
+  },
+  outlineColor: {
+    default: 'focus-ring',
+    forcedColors: 'Highlight'
+  },
+  padding: {
+    orientation: {
+      horizontal: {
+        default: 16
+      }
+    }
+  },
+  position: 'relative'
+});
+
+const selectBoxContentStyle = style({
+  alignContent: 'center',
+  alignItems: 'center',
+  display: 'grid',
+  gap: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: 0,
+          M: size(2)
+        },
+        vertical: size(8)
+      }
+    }
+  },
+  gridTemplateAreas: {
+    orientation: {
+      horizontal: ['icon label', 'icon description'],
+      vertical: ['. icon .', '. label .']
+    }
+  },
+  gridTemplateColumns: {
+    orientation: {
+      horizontal: '48px 1fr'
+    }
+  },
+  height: 'full',
+  justifyItems: {
+    orientation: {
+      horizontal: 'flex-start',
+      vertical: 'center'
+    }
+  },
+  paddingStart: {
+    orientation: {
+      horizontal: size(12),
+      vertical: 0
+    }
+  },
+  width: 'auto'
+});
+
+const selectBoxIconStyle = style({
+  color: {
```

Comment:
see https://github.com/adobe/react-spectrum/blob/e92ccd7568db87ed15b828b9e0b4a7de5dababd1/packages/%40react-spectrum/s2/src/InlineAlert.tsx#L124 for how to generally color icons